README.md

```diff
@@ -5,105 +5,127 @@
 
-The `ballerinax/ibm.ibmmq` library provides an API to connect to an IBM MQ server using Ballerina.
+[IBM MQ](https://www.ibm.com/products/mq) is a powerful messaging middleware platform designed for facilitating reliable
+communication between disparate systems and applications. IBM MQ ensures the secure and orderly exchange of messages
+asynchronously, decoupling senders and receivers for efficient and scalable communication. It supports both
+point-to-point and publish/subscribe messaging models via queues and topics.
 
-This library is created with minimal deviation from the IBM MQ java client API to make it easy for the developers who are used to working with the IBM MQ java client.
+The `ballerinax/ibm.ibmmq` package provides an API to connect to an IBM MQ server using Ballerina.
 
-Currently, the following IBM MQ API Classes are supported through this package.
+## Setup guide
 
-- QueueManager
-- Queue
-- Destination (Queue, Topic)
-- Message
+To use the Ballerina IBM MQ connector, you need to have an IBM MQ instance running or possess an IBM MQ cloud account.
+For setting up IBM MQ locally, you can refer to the [IBM MQ official documentation](https://www.ibm.com/docs/en/ibm-mq/9.3?topic=migrating-installing-uninstalling).
+Alternatively, to use IBM MQ on the cloud, [sign up](https://cloud.ibm.com/registration) for an IBM MQ cloud account.
 
-### IBM MQ queue
+### Create a queue
 
-IBM MQ queues are used for point-to-point messaging. In point-to-point messaging, a producer application sends a message to a queue, and a single consumer application retrieves the message from the queue.
+1. Log into IBM MQ console. If you are running an IBM MQ server locally you can navigate to
+   [https://localhost:9443/ibmmq/console](https://localhost:9443/ibmmq/console) URL in your browser to access the IBM MQ console.
+2. Click on the `Create a queue` link.
 
-#### Producer
+![Create a queue](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-ibm.ibmmq/main/docs/setup/resources/select-create-queue.png)
+
+3. Select the queue type.
+
+![Select the queue type](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-ibm.ibmmq/main/docs/setup/resources/select-queue-type.png)
+
+4. Go back to the home page and click on the `Manage` link on the sidebar.
+
+![Click on Manage](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-ibm.ibmmq/main/docs/setup/resources/click-manage-link.png)
+
+5. Navigate to `Events` tab.
+
+![Navigate to Events tab](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-ibm.ibmmq/main/docs/setup/resources/navigate-to-events-tab.png)
+
+6. Click on `Create`.
+
+![Click on create](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-ibm.ibmmq/main/docs/setup/resources/click-on-create.png)
```

Comment:
shall we crop the screenshots to remove the menu bar in the browser?

Reply:
Fixed with [77ff4e23d6c4be861035a1dd16cf133cfba89d19](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/pull/37/commits/77ff4e23d6c4be861035a1dd16cf133cfba89d19)